### PR TITLE
Fix deadlock in system tables during server startup

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed deadlock during server startup after update with changes in structure of system log tables.

### CODE REVIEW

**Comment by alesapin** [comment]
It's already tested by `test_version_update_after_mutation` integration test. But we run them with retries and sometimes it helps.

**Comment by alesapin** [comment]
Deadlock happens here https://github.com/clickhouse/ClickHouse/blob/master/src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp#L186

File: src/Interpreters/Context.h [hunk 1]
```diff
@@ -541,14 +541,12 @@ class Context
...
 
-    DiskSelectorPtr getDiskSelector() const;
 
     /// Provides storage disks
     DiskPtr getDisk(const String & name) const;
-    DiskPtr getDefaultDisk() const { return getDisk("default"); }
-
-    StoragePolicySelectorPtr getStoragePolicySelector() const;
 
+    std::map<String, StoragePolicyPtr> getPoliciesMap() const;
```
Comment (KochetovNicolai) [inline src/Interpreters/Context.h:548]: Let's use `using` for this maps

**Review by KochetovNicolai — APPROVED** [review]
LGTM

**Comment by alesapin** [comment]
Error expected

**Comment by alesapin** [comment]
All affected tests passed.

**Comment by nikitamikhaylov** [comment]
Decided not to cherry pick it because of "volumes" and etc

**Pull request merged**